### PR TITLE
Closes #86: Rewrite PDB Api importer

### LINKED ISSUE #86: Add possibility of retrieving data from pdb rest json api

### PR DESCRIPTION
**Pull request by wtr-icm**

### CODE REVIEW

File: src/main/java/pl/edu/icm/pl/mxrdr/extension/importer/pdb/PdbApiCaller.java [hunk 1]
```diff
@@ -1,117 +1,100 @@
...
+        }
+        PdbDataContainer pdbDataContainer = new PdbDataContainer();
+        pdbDataContainer.init(structureData);
+        return structureData;
     }
 
     // -------------------- PRIVATE --------------------
 
-    private <T> T executeForObject(Function<URI, Request> request, URI uri, Class<T> resultType) {
+    private <T> T getAndMap(URI uri, Class<T> resultType) {
```
Comment (MrK191) [inline src/main/java/pl/edu/icm/pl/mxrdr/extension/importer/pdb/PdbApiCaller.java:72]: Maybe 'requestData'? getAndMap is kinda confusing to me.

File: src/main/java/pl/edu/icm/pl/mxrdr/extension/importer/pdb/PdbApiCaller.java [hunk 2]
```diff
@@ -1,117 +1,100 @@
...
+
+        URI entryEndpointUri = buildUri(baseUrl, ENTRY_PATH, structureId);
+        EntryData entryData = getAndMap(entryEndpointUri, EntryData.class);
+        StructureData structureData = new StructureData(entryData);
+
+        Integer polymerEntityCount = structureData.getEntryData()
+                .getRcsbEntryInfo()
+                .getPolymerEntityCount();
+
+        if (polymerEntityCount == null || polymerEntityCount < 1) {
```
Comment (MrK191) [inline src/main/java/pl/edu/icm/pl/mxrdr/extension/importer/pdb/PdbApiCaller.java:56]: I'm thinking if we should have some kind of documentation for this kind of checks or is it enough to just learn pdb domain, what do you think?

**Review by MrK191 — CHANGES_REQUESTED** [review]

File: src/main/java/pl/edu/icm/pl/mxrdr/extension/importer/pdb/PdbApiCaller.java (hunk 1 above)
Comment (wtr-icm) [reply src/main/java/pl/edu/icm/pl/mxrdr/extension/importer/pdb/PdbApiCaller.java:72]: Maybe `requestDataAndDeserialize`?

File: src/main/java/pl/edu/icm/pl/mxrdr/extension/importer/pdb/PdbApiCaller.java (hunk 2 above)
Comment (wtr-icm) [reply src/main/java/pl/edu/icm/pl/mxrdr/extension/importer/pdb/PdbApiCaller.java:56]: I'll write a comment, but generally I think we do not have consistent strategy for such issues and that's something we can discuss tomorrow on daily.

**Review by MrK191 — APPROVED** [review]

**Pull request merged**